src/vellum/workflows/nodes/displayable/tool_calling_node/tests/test_node.py

```diff
@@ -140,3 +146,65 @@ def test_deployment_definition_release_tag_defaults_to_latest():
 
     # THEN the release_tag should default to "LATEST"
     assert deployment_config.release_tag == "LATEST"
+
+
+def test_tool_calling_node_with_user_provided_chat_history_block(vellum_adhoc_prompt_client):
+    """
+    Test that ToolCallingNode with user-provided chat history block merges user and node messages.
+    """
+
+    # GIVEN a ToolCallingNode with a user-provided chat history block
+    user_chat_history_block = VariablePromptBlock(
+        block_type="VARIABLE",
+        input_variable="chat_history",
+        state=None,
+        cache_config=None,
+    )
+
+    class TestToolCallingNode(ToolCallingNode):
+        ml_model = "gpt-4o-mini"
+        blocks = [user_chat_history_block]
+        functions = [first_function]
+        prompt_inputs = {"chat_history": [ChatMessage(role="USER", text="Hello from user")]}
+        max_prompt_iterations = 1
+
+    def generate_prompt_events(*args: Any, **kwargs: Any) -> Iterator[Any]:
+        execution_id = str(uuid4())
+        events = [
+            InitiatedExecutePromptEvent(execution_id=execution_id),
+            FulfilledExecutePromptEvent(
+                execution_id=execution_id,
+                outputs=[StringVellumValue(value="Hello! I can help you.")],
+            ),
+        ]
+        yield from events
+
+    vellum_adhoc_prompt_client.adhoc_execute_prompt_stream.side_effect = generate_prompt_events
+
+    # AND a state
+    state = BaseState()
+
+    # WHEN the ToolCallingNode runs
+    node = TestToolCallingNode(state=state)
+    list(node.run())
+
+    # THEN the API should be called with the correct blocks
+    mock_api = vellum_adhoc_prompt_client.adhoc_execute_prompt_stream
+    assert mock_api.call_count >= 1
+
+    # AND the blocks should include the user-provided chat_history block
+    call_kwargs = mock_api.call_args.kwargs
+    blocks = call_kwargs["blocks"]
+
+    chat_history_blocks = [
+        block for block in blocks if block.block_type == "VARIABLE" and block.input_variable == "chat_history"
+    ]
+    assert len(chat_history_blocks) == 1
+
+    # AND the input_values should include the user's chat history
+    input_values = call_kwargs["input_values"]
+    chat_history_inputs = [
+        input_val for input_val in input_values if hasattr(input_val, "key") and input_val.key == "chat_history"
+    ]
+    assert len(chat_history_inputs) == 1
+    assert chat_history_inputs[0].value == []
```

Comment:
This value should be equal to `[ChatMessage(role="USER", text="Hello from user")]`. Make this update and then update out implementation to accommodate for this.